Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
       PEP written by Marc-André Lemburg.
 
 
+.. We use the following practises for SQL code:
+   - UPPERCASE for keywords
+   - snake_case for schema
+   - single quotes for string literals
+   - singular for table names
+   - if needed, use double quotes for table and column names
+
 .. _sqlite3-tutorial:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating a database of Monty Python movies.
+It assumes a fundamental understanding of database concepts,
+including `cursors`_ and `transactions`_.
+
+First, you'll need to create a new database to hold the movie data
+and open a database connection to allow :mod:`!sqlite3` to work with it.
+Call the :func:`sqlite3.connect` function to
+to create a connection to a database :file:`tutorial.db`
+in the current working directory,
+implicitly creating it if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+you'll use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
-
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+Now you'll create a database table ``movie`` with columns for title,
+release year, and review score.
+For simplicitly, just use column names in the table declaration:
+thanks to the `flexible typing`_ feature of SQLite,
```

Comment:
```suggestion
For simplicity, just use column names in the table declaration —
thanks to the `flexible typing`_ feature of SQLite,
```

or

```suggestion
For simplicity, just use column names in the table declaration;
thanks to the `flexible typing`_ feature of SQLite,
```

* Fix spelling typo
* A parenthetical (em dash/semicolon) instead of colon makes more sense here, IMO

Reply:
FWIW in Sphinx you can create en and em dashes using `--` and `---` respectively, without having to include them directly in the rst.

Reply:
I see—is there a reason to avoid the literal characters in the source? At least for me its a lot simpler to just enter the normal characters directly (via Alt-0150/0151 or Option (Shift) -) since that's what works everywhere else, vs. trying to remember a bespoke RST-specific method.

Reply:
I prefer the `--` variant, as it is easier to spot when reading the raw text.

Reply:
> I see—is there a reason to avoid the literal characters in the source?

Back then, non-ASCII characters used to break the PDF build, but I think we are past those dark times.  There is still value in keeping the docs ASCII-only -- not everyone expects, recognizes, and is able to type non-ASCII characters easily, which might either lead to inconsistencies or accidentally replacing a `—` with a `-`.

> I prefer the `--` variant, as it is easier to spot when reading the raw text.

On the other hand, it's very easy to mix `--` and `---` up, since most people use `--` when they want an em dash, whereas Sphinx wants `---` and uses `--` for en dash. :upside_down_face: